controllers/hpa_util.go

```diff
@@ -21,19 +21,79 @@ import (
 	"fmt"
 	"reflect"
 
-	autoscalingv1alpha1 "github.com/gardener/hvpa-controller/api/v1alpha1"
+	autoscalingv2 "k8s.io/api/autoscaling/v2"
 	autoscaling "k8s.io/api/autoscaling/v2beta1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/types"
 	errorsutil "k8s.io/apimachinery/pkg/util/errors"
 	"k8s.io/client-go/util/retry"
+	kubernetesinternalautoscaling "k8s.io/kubernetes/pkg/apis/autoscaling"
+
+	autoscalingv1alpha1 "github.com/gardener/hvpa-controller/api/v1alpha1"
 )
 
 // TODO: use client library instead when it starts to support update retries
 //
 //	see https://github.com/kubernetes/kubernetes/issues/21479
 type updateHpaFunc func(hpa *autoscaling.HorizontalPodAutoscaler)
+type updateHpaV2Func func(hpa *autoscalingv2.HorizontalPodAutoscaler)
+
+func (r *HvpaReconciler) Convert_v2beta1_HorizontalPodAutoscalerList_To_v2(in *autoscaling.HorizontalPodAutoscalerList) (*autoscalingv2.HorizontalPodAutoscalerList, error) {
```

Comment:
This function should document whether its output may contain references to parts of the input object, or if the output is guaranteed to be an independent (deep) copy.